bundles/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/internal/handler/FeedHandler.java

```diff
@@ -113,13 +128,91 @@ private void refreshFeedState() {
         boolean feedUpdated = updateFeedIfChanged(feed);
 
         if (feedUpdated) {
+            publishDynamicChannelsIfLinked();
+
             List<Channel> channels = getThing().getChannels();
             for (Channel channel : channels) {
                 publishChannelIfLinked(channel.getUID());
             }
         }
     }
 
+    private void publishDynamicChannelsIfLinked() {
+        List<SyndEntry> entries = currentFeedState.getEntries()
+                .stream()
+                .limit(numberOfEntries.intValue())
+                .collect(Collectors.toList());
+
+        createDynamicChannels(numberOfEntries.intValue());
```

Comment:
Please move the creation of channels to `initialize` and also check if the number of channels present is higher than the number of channels configured and remove them if necessary.